test/library/standard/Random/RandomStreamInterface/choiceTestDomain.chpl

```diff
@@ -0,0 +1,131 @@
+use Random, Map;
```

Comment:
Can you integrate these test cases into the existing test for choice?

Reply:
> Can you integrate these test cases into the existing test for choice?

@ben-albrecht I thought it would be cleaner to keep the tests in separate files :sweat_smile: .

Though if we keep the tests in the same file, the `testx` (where x can be domain, range o array will require changes, it will involve some conditional statements depending on the argument being an array, domain or range. Some print statements will also need to be standardized without  explicitly mentioning the `arg` type.

Reply:
Well, the test programs have a lot of common functionality that gets duplicated by splitting them up. By merging them and adding the conditional statements needed, we can improve the maintainability of the tests.

We could also keep the tests in separate files and place the common code into a non-tested file like `choiceTestUtils.chpl` with a  `choiceTestUtils.notest`.

Reply:
> Well, the test programs have a lot of common functionality that gets duplicated by splitting them up. By merging them and adding the conditional statements needed, we can improve the maintainability of the tests.

Yes! agreed. 

> We could also keep the tests in separate files and place the common code into a non-tested file like `choiceTestUtils.chpl` with a `choiceTestUtils.notest`.

Yeah, this is a good alternative :smile:

Reply:
@ben-albrecht the `test` function code that is common between  `range`, `array` and `domain` do a comparison between the `actualRatios` and the `expectedRatios` obtained from the `prob` array. There is no functionality that requires specific elements of the `array` other than a couple of print statements, which I think can be replaced by the corresponding domain elements.

```chapel
  if debug {
    writeln('Counts for array: ', arr);
    for value in counts {
      writeln(value, ' : ', counts[value]/trials:real);
    }
  }
```
and 

```
    write('[', arr, '], ');
```
So, I was thinking of putting the `test` proc in a utils file as you suggested and modifying it to expect only a `domain` as an argument for all three cases.

What do you think?

Thanks!

Reply:
@Aniket21mathur - Yes, that sounds like a good plan to me. Thanks for confirming.

Reply:
@ben-albrecht I am having some second thoughts on it, if we go the way I described, the `range` and `array` overloads will not be utilized in tests. Also, the tests currently in `master` just `Confirm that resulting ratios are within 0.05 of expected ratios`, keeping that in mind, do we need seperate tests for `range` , `array` and `domain`? Ultimately we are passing a domain to the `_choice` function. Though the error tests will be different for sure.

Reply:
Good point.

Is it possible to make `testArray` generic so that it will accept any range, domain, or array and pass it off to `choice` without doing anything specific to the type?

If necessary, you could split the validation code out into separate overloaded functions to validate the results for each type (it's not clear to me if that's necessary or not though - at a glance).